asapdiscovery/data/pdb.py

```diff
@@ -30,60 +31,92 @@ def download_pdb_structure(pdb_id: str, directory: str):
         The PDB ID of interest.
     directory: str or Path, default=user_cache_dir
         The directory for saving the downloaded structure.
+    file_format: str, optional
+        Indicates whether you would like to download the entry in pdb ("pdb") or cif format ("cif"),
+        or the first biological assembly in cif format ("cif1"). Defaults to "pdb".
 
     Returns
     -------
-    : Path or False
-        The path to the the downloaded file if successful, else False.
+    file_path : Path or False
+        The path to the downloaded file if successful, else False.
     """
     from .utils import download_file
     import os
 
-    # check for structure in PDB format
-    url = f"https://files.rcsb.org/download/{pdb_id}.pdb"
-    pdb_path = os.path.join(directory, f"rcsb_{pdb_id.upper()}.pdb")
-    if not os.path.exists(pdb_path):
-        print("Downloading PDB entry in PDB format ...")
-        if download_file(url, pdb_path):
-            return pdb_path
-    else:
-        return pdb_path
-
-    # check for structure in CIF format
-    url = f"https://files.rcsb.org/download/{pdb_id}.cif"
-    cif_path = os.path.join(directory, f"rcsb_{pdb_id.upper()}.cif")
-    if not os.path.exists(cif_path):
-        print("Downloading PDB entry in CIF format ...")
-        if download_file(url, cif_path):
-            return cif_path
+    url_base_str = (
+        f"https://files.rcsb.org/download/"  # base str to use for URLs
+    )
+    # Dictionary with allowed formats and their upstream basenames
+    format_to_basename = {
+        "pdb": f"{pdb_id}.pdb",
+        "cif": f"{pdb_id}.cif",
+        "cif1": f"{pdb_id}-assembly1.cif",
+    }
+
+    allowed_types = format_to_basename.keys()
+    # Make sure pdb_type can be handled
+    if file_format not in allowed_types:
+        raise NotImplementedError(
+            f"pdb_type expected to be one of {allowed_types}, not '{file_format}'"
+        )
+
+    basename = format_to_basename[file_format]
+    local_path = os.path.join(directory, f"rcsb_{basename}")
+    # Download only if it doesn't exist locally
+    if not os.path.exists(local_path):
+        url = f"{url_base_str}{basename}"
+        response = download_file(url, local_path)
+        if response.status_code == 200:
+            result = local_path
+        elif response.ok:
+            raise requests.HTTPError(
+                (
+                    f"Received status code {response.status_code}, "
+                    "file not downloaded."
+                )
+            )
+        else:
+            response.raise_for_status()
```

Comment:
@kaminow Thanks for this improvement. Just one question, why is it that we have to check for `200` explicitly? I guess it can return `response.ok==True` even without downloading it, what's exactly happening in that case then? Can you clarify that for me? Thanks.

Reply:
`response.ok` will return `True` for any code < 400, which I think includes codes that mean the file wasn't downloaded. I'm certainly no networking expert though so I could be wrong about this

Reply:
I see, thanks for clarifying!